### DISCUSSION TITLE
Handle large crowds that stuck at a corner

### DISCUSSION THREAD

**Discussion by eriksachse** [recastnavigation/recastnavigation#675] [Q&A]
Ayyyyyo, 
when using detour with a large crowd simulation, they tend to get stuck at the corner. My workaround would be that if velocity is below a certain treshold, change the `collisionQueryRange` so that the agents loose up a little.

I'm open for any other suggestions. See video for reference 👽

https://github.com/recastnavigation/recastnavigation/assets/24212703/f2599d9c-6a8f-46b7-9ab5-3c2dfe67bde8

**Comment by grahamboree** [comment]
Conceptually what's happening is that lots of agents are bumping up and fighting each other to get to the same waypoint position.  They're also not evaluating whether the next waypoint is within line of sight and skipping to that one.  I'm not sure where this issue is occurring but it's possible that Detour doesn't have a super robust path-following behavior.  I'd have to dig more into it.  Maybe a stretch to ask but is there any kind of minimal repro case you could share?

**Comment by ZA139** [comment]
Is there any possible nav to different waypoints which separated by one nav agent size?
Just like the hitman did modify the corner navpoint. URL: https://www.gdcvault.com/play/1015315/Crowds-in-Hitman
